File: lib/app/user_and_seller/model/liked_product.dart
class likedProduct {
  int lid;
  int pid;
  String imageUrl;
  String name;
  String description;
  double price;
  int categoryId;
  int sellerId;
  double gst;
  String emailId;

  likedProduct({
    required this.lid,
   required this.pid,
   required this.imageUrl,
   required this.name,
   required this.description,
   required this.price,
   required this.categoryId,
   required this.sellerId,
   required this.gst,
   required this.emailId,
  });

  factory likedProduct.fromJson(Map<String, dynamic> json) {
    return likedProduct(
      lid: int.parse(json['lid']),
      pid: int.parse(json['pid']),
      imageUrl: json['imageurl'],
      name: json['name'],
      description: json['description'],
      price: double.parse(json['price']), // Convert to double
      categoryId: int.parse(json['categoryid']),
      sellerId: int.parse(json['seller_id']),
      gst: double.parse(json['gst']), // Convert to double
      emailId: json['email_id'],
    );
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['lid'] = this.lid;
    data['pid'] = this.pid;
    data['imageurl'] = this.imageUrl;
    data['name'] = this.name;
    data['description'] = this.description;
    data['price'] = this.price;
    data['categoryid'] = this.categoryId;
    data['seller_id'] = this.sellerId;
    data['gst'] = this.gst;
    data['email_id'] = this.emailId;
    return data;
  }
}

File: php/liked_product.php
<?php

include "dbUpload.php";

// Retrieve values sent from Flutter through POST request
$imageurl = $_POST['imageurl'];
$name = $_POST['name'];
$description = $_POST['description'];
$price = $_POST['price'];
$categoryid = $_POST['categoryid'];
$seller_id = $_POST['seller_id'];
$gst = $_POST['gst'];
$email_id = $_POST['email_id'];
$pid = $_POST['pid']; 

// SQL query to check if the row with the same pid and email already exists
$check_sql = "SELECT * FROM likedproduct WHERE pid='$pid' AND email_id='$email_id'";
$check_result = $conn->query($check_sql);

if ($check_result->num_rows > 0) {
    // Row with the same pid and email already exists
    $response = array("status" => "error", "message" => "exists");
    echo json_encode($response);
} else {
    // Row does not exist, proceed with inserting the new row
    $insert_sql = "INSERT INTO likedproduct (pid, imageurl, name, description, price, categoryid, seller_id, gst, email_id) 
                   VALUES ('$pid', '$imageurl', '$name', '$description', '$price', '$categoryid', '$seller_id', '$gst', '$email_id')";

    if ($conn->query($insert_sql) === TRUE) {
        $response = array("status" => "success", "message" => "success");
        echo json_encode($response);
    } else {
        $response = array("status" => "error", "message" => "Error: " . $insert_sql . "<br>" . $conn->error);
        echo json_encode($response);
    }
}

// Close connection
$conn->close();

?>
